lib/purview/databases/base.rb

```diff
@@ -446,6 +466,11 @@ def next_window(connection, table, timestamp)
         )
         max = min + table.window_size
```

Comment:
can the setting of these mins and maxs be put into separate methods?  it's incredibly hard to follow the logic through this method - max is set to min + table.window_size, then it's checked against the highest min and possibly set to min + baseline_window, then inside the Window instantiation, it's checked again against now and set to either now or max.  can this be cleaned up a bit?